### PR TITLE
POC -Data schema validation

### PR DESCRIPTION
**Pull request by pjlee11**
_Create a validator which using our schema validates the data feed._

* [x] Fixes https://github.com/bbc/simorgh/issues/91
* [x] Up-to-date with latest - `git pull --rebase origin latest`
* [x] Runs locally - `npm run dev` & [http://localhost:7080/](http://localhost:7080/)
* [x] Tests added for new features
* [x] Tests pass - `npm test`
* [x] E2E tests pass - `npm run test:e2e` (requires server running see [README](https://github.com/bbc/simorgh#tests) for details)

* [ ] Test engineer approval

### CODE REVIEW

**Comment by pjlee11** [comment]
🎉 Output following commit 16bc5fe

```
Validating block: article
----------------------------------------------------------------
- Valid type of object
- Required values successfully found:
  - locator
  - publishedDateTime
  - lastUpdatedDateTime
  - passport
  - seoHeadline
  - metaDescription
  - model

Validating property :article:locator
- Valid type of string

Validating property :article:passport
- Valid type of object
- Required values successfully found:
  - home
  - language
  - articleType

Validating property :article:passport:language
- Valid type of string

Validating property :article:passport:home
- Valid type of string

Validating property :article:passport:articleType
- Valid enum of news
- Valid type of string

Validating property :article:seoHeadline
TYPE OF NULL

Validating property :article:publishedDateTime
- Valid type of string

Validating property :article:lastUpdatedDateTime
- Valid type of string

Validating property :article:metaDescription
TYPE OF NULL

Validating property :article:model


Validating block: headline
----------------------------------------------------------------
- Valid type of object
- Required values successfully found:
  - blockId
  - type
  - model

Validating property :article:model:headline:blockId
- Valid type of string

Validating property :article:model:headline:type
- Valid enum of headline
- Valid type of string

Validating property :article:model:headline:model


Validating block: text
----------------------------------------------------------------
- Valid type of object
- Required values successfully found:
  - blockId
  - type
  - model

Validating property :article:model:headline:model:text:blockId
- Valid type of string

Validating property :article:model:headline:model:text:type
- Valid enum of text
- Valid type of string

Validating property :article:model:headline:model:text:model


Validating block: paragraph
----------------------------------------------------------------
- Valid type of object
- Required values successfully found:
  - blockId
  - type
  - model

Validating property :article:model:headline:model:text:model:paragraph:blockId
- Valid type of string

Validating property :article:model:headline:model:text:model:paragraph:type
- Valid enum of paragraph
- Valid type of string

Validating property :article:model:headline:model:text:model:paragraph:model


Validating block: text
----------------------------------------------------------------
- Valid type of object
- Required values successfully found:
  - blockId
  - type
  - model

Validating property :article:model:text:blockId
- Valid type of string

Validating property :article:model:text:type
- Valid enum of text
- Valid type of string

Validating property :article:model:text:model


Validating block: paragraph
----------------------------------------------------------------
- Valid type of object
- Required values successfully found:
  - blockId
  - type
  - model

Validating property :article:model:text:model:paragraph:blockId
- Valid type of string

Validating property :article:model:text:model:paragraph:type
- Valid enum of paragraph
- Valid type of string

Validating property :article:model:text:model:paragraph:model
- Valid type of object

Validating property :article:model:blocks

Validating property :article:model:blocks:blockId
TYPE OF NULL

Validating property :article:model:blocks:type
TYPE OF NULL

Validating property :article:model:blocks:model
TYPE OF NULL

Validation complete!
```

File: package.json [hunk 1]
```diff
@@ -69,6 +70,7 @@
     "eslint-plugin-json": "^1.2.0",
     "eslint-plugin-jsx-a11y": "^6.0.3",
     "eslint-plugin-prettier": "^2.6.1",
+    "fs": "0.0.1-security",
```
Comment (jtart) [inline package.json:73]: Remove this lib - `fs` is a native JS lib.

**Review by jtart — CHANGES_REQUESTED** [review]
Can you split this PR up into schema changes and the data validation logic? The schema changes are substantial and should not be included in this PR. This also means that the PR is difficult to code review as it is complex - we are aiming to keep PRs as atomic as possible.

**Comment by pjlee11** [comment]
@jtart happy to split out the schema changes, however its worth noting the required changes to the schema would have never been noticed until the validation logic was actually using the schema so it was inevitable that changes were going to occur. 
In terms on the validation logic itself I think splitting that out would be a mistake as it would be misleading it terms of how the final validator would look. This is because of the way the schema and data recurse if I were to split the logic out into PRs each PR will naturally contain future optimisation as the validator now has full logic. It would be like seeing the whole puzzle before starting to put each piece together.

File: package.json (hunk 1 above)
Comment (pjlee11) [reply package.json:73]: Will do

**Comment by jtart** [comment]
@pjlee11 I think splitting out the schema changes and the schema validation is sensible to split up. If it does not make sense to split up the validation into smaller PRs then it might be fine to keep as a single PR.

**Comment by pjlee11** [comment]
This PR is going to be split into smaller, more reviewable PRs

**Comment by jamesbhobbs** [comment]
@pjlee11 should this still be open? I think it's been split into other PRs and that we should comment those PRs here and close.

**Comment by pjlee11** [comment]
Closed in favour of https://github.com/bbc/simorgh/pull/271 and https://github.com/bbc/simorgh/pull/273 and part 3 (PR not yet made)

**Pull request closed without merging**